test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/drools/DroolsSessionFailoverIntegrationTest.java

```diff
@@ -0,0 +1,162 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(DroolsSessionFailoverIntegrationTest.class);
+    private static final String SMART_ROUTER_ID = "test-kie-router";
+
+    private static final String LIST_NAME = "list";
+    private static final String LIST_OUTPUT_NAME = "output-list";
+    private static final String KIE_SESSION = "defaultKieSession";
+
+    private KieServerMgmtControllerClient kieServerMgmtControllerClient;
+    private KieServicesClient smartRouterServicesClient;
+    private RuleServicesClient smartRouterRuleServiceClient;
+
+    private KieServicesClient kieServerClient;
+
+    private static KieCommands commandsFactory = KieServices.Factory.get().getCommands();
+
+    private static final String FAST_RULE = "Fast rule executed";
+    private static final String SLOW_RULE = "Slow rule executed";
+
+    @Override
+    protected WorkbenchRuntimeSmartRouterKieServerDatabaseScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
+        return deploymentScenarioFactory.getWorkbenchRuntimeSmartRouterKieServerDatabaseScenarioBuilder()
+                .withExternalMavenRepo(MavenConstants.getMavenRepoUrl(), MavenConstants.getMavenRepoUser(), MavenConstants.getMavenRepoPassword())
+                .withSmartRouterId(SMART_ROUTER_ID)
+                .build();
+    }
+
+    @Before
+    public void setUp() {
+        MavenDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/rule-project").getFile());
+
+        kieServerMgmtControllerClient = KieServerControllerClientProvider.getKieServerMgmtControllerClient(deploymentScenario.getWorkbenchRuntimeDeployment());
+        smartRouterServicesClient = KieServerClientProvider.getSmartRouterClient(deploymentScenario.getSmartRouterDeployment(), deploymentScenario.getKieServerDeployment().getUsername(), deploymentScenario.getKieServerDeployment().getPassword(), TimeUnit.MINUTES.toMillis(10));
+        smartRouterRuleServiceClient = smartRouterServicesClient.getServicesClient(RuleServicesClient.class);
+
+        kieServerClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
+
+        deploymentScenario.getSmartRouterDeployment().setRouterTimeout(Duration.ofMinutes(5));
+    }
+
+    @Test
+    @Ignore("Activate when RHBPMS-5044 is done.")
+    public void executeSimpleRuleFailoverTest() throws InterruptedException {
+        logger.debug("Register Kie Container to Kie Server");
+        KieServerInfo serverInfo = kieServerClient.getServerInfo().getResult();
+        kieServerMgmtControllerClient.saveContainerSpec(serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, RULE_PROJECT_NAME, RULE_PROJECT_VERSION, KieContainerStatus.STARTED);
+        KieServerClientProvider.waitForContainerStart(deploymentScenario.getKieServerDeployment(), CONTAINER_ID);
```

Comment:
Please add waiting for container registration into Smart router as it is used later in the code - https://github.com/kiegroup/kie-cloud-tests/blob/c2d5e7dac45a5a3a2d3cf61187b9affdadfb2550/test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/emptymanagedkieserver/EmptyManagedKieServerConnectionIntegrationTest.java#L91

Reply:
Added. See the next line.